src/coreclr/gc/gcinterface.h

```diff
@@ -30,12 +30,19 @@ typedef void enum_alloc_context_func(gc_alloc_context*, void*);
 // Callback passed to CreateBackgroundThread.
 typedef uint32_t (__stdcall *GCBackgroundThreadFunction)(void* param);
 
+template<typename ELEMENT,typename TRAITS>
+class SetSHash;
+
+template<typename ELEMENT>
+class PtrSetSHashTraits;
+
 // Struct often used as a parameter to callbacks.
 typedef struct
 {
     promote_func*  f;
     ScanContext*   sc;
     CrawlFrame *   cf;
+    SetSHash<Object**, PtrSetSHashTraits<Object**> > *pScannedSlots;
```

Comment:
Can we add a field here - will it break the interface compatibility?

Reply:
This structure is used only on the VM side, so it seems it should be ok.